crates/ink/src/codegen/dispatch/execution.rs

```diff
@@ -78,10 +75,10 @@ where
             //
             // We need to revert the state of the transaction.
             ink_env::return_value::<
-                <private::Seal<R> as ConstructorReturnType<Contract>>::ReturnValue,
+                <private::Seal<R> as ConstructorReturnType<Contract>>::Error,
             >(
                 ReturnFlags::default().set_reverted(true),
-                result.return_value(),
+                result.error_value(),
```

Comment:
instead of `error_value` you could (and should) be using the `Err(error_value)` that you just matched against.